ietf/api/views.py

```diff
@@ -232,6 +233,153 @@ def err(code, text):
+    except jsonschema.exceptions.ValidationError as err:
+        return _http_err(400, f"JSON schema error at {err.json_path}: {err.message}")
+    except Exception:
+        return _http_err(400, "Invalid request format")
+
+    # Validate consistency
+    # - if receive multiple records they should be for same meeting, same person (email)
+    if len(payload) > 1:
+        if len(set([r['meeting'] for r in payload])) != 1:
+            return _http_err(400, "Different meeting values")
+        if len(set([r['email'] for r in payload])) != 1:
+            return _http_err(400, "Different email values")
+
+    # Validate meeting
+    number = payload[0]['meeting']
+    try:
+        meeting = Meeting.objects.get(number=number)
+    except Meeting.DoesNotExist:
+        return _http_err(400, "Invalid meeting value: '%s'" % (number, ))
+
+    # Validate email
+    email = payload[0]['email']
+    try:
+        validate_email(email)
+    except ValidationError:
+        return _http_err(400, "Invalid email value: '%s'" % (email, ))
+
+    # handle cancelled. there will be only one record
+    if payload[0]['cancelled']:
+        reg = MeetingRegistration.objects.filter(
+            meeting__number=number,
+            email=email,
+            reg_type=payload[0]['reg_type'],
+            ticket_type=payload[0]['ticket_type']).first()
+        if reg:
+            reg.delete()
+        return HttpResponse('Success', status=202, content_type='text/plain')
+
+    # get person
+    person = Person.objects.filter(email__address=email).first()
+    if not person:
+        log.log(f"api_new_meeting_registration_v2 no Person found for {email}")
+
+    # get existing records if any
+    regs = MeetingRegistration.objects.filter(meeting__number=number, email=email)
+    pks = [r.pk for r in regs]
+
+    for registration in payload:
+        new_reg = MeetingRegistration(
+            meeting=meeting,
+            first_name=registration['first_name'],
+            last_name=registration['last_name'],
+            affiliation=registration['affiliation'],
+            country_code=registration['country_code'],
+            person=person,
+            email=email,
+            reg_type=registration['reg_type'],
+            ticket_type=registration['ticket_type'],
+            checkedin=registration['checkedin'])
+
+        # update any existing records if there are any
+        new_reg.pk = _safe_pop(pks)
+        if new_reg.pk:
+            log.log(f"Updating MeetingRegistration record for meeting:{meeting} email:{email}")
+        else:
+            log.log(f"New MeetingRegistration record for meeting:{meeting} email:{email}")
+        new_reg.save()
+
+        # removed account creation email. Registration requires Datatracker account
+
+        # handle nomcom volunteer
+        if registration['is_nomcom_volunteer'] and person:
+            try:
+                nomcom = NomCom.objects.get(is_accepting_volunteers=True)
```

Comment:
Does the registration system check to see if a nomcom is accepting volunteers before offering the choice to someone?

Reply:
It does not. It's another level of integration that could be added. My understanding was that AMS would coordinate with the nomcom chair. When prepping to open registration if nomcom is accepting volunteers I would ensure the volunteer question appears on the reg form.